### PR TITLE
Add extra Kompose documentation

### PR DESCRIPTION
**Pull request by cdrage**
Create additional files for Kompose documentation including
architecture, development, a more extensive index and a user guide.

This also bumps Kompose up to the top on tools page to be symmetrical to
the what's listed on the side navbar.

I have assigned myself to each doc in order to get updates / any
questions regarding the updating of documentation.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/1607)

### CODE REVIEW

**Comment by cdrage** [comment]
ping @ngtuna @tkral @janetkuo

File: docs/tools/kompose/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+| Key                  | Value                               |
+|----------------------|-------------------------------------|
+| kompose.service.type | nodeport / clusterip / loadbalancer |
+
+
+For example:
+
+```yaml
+---
```
Comment (janetkuo) [inline docs/tools/kompose/user-guide.md]: remove `---`

File: docs/tools/kompose/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+`kompose` supports Kompose-specific labels within the `docker-compose.yml` file in order to explicitly imply a service type upon conversion.
+
+The currently supported options are:
+
+| Key                  | Value                               |
+|----------------------|-------------------------------------|
+| kompose.service.type | nodeport / clusterip / loadbalancer |
+
+
+For example:
```
Comment (janetkuo) [inline docs/tools/kompose/user-guide.md]: Somehow this part seems unfinished. Maybe change "for example" to "Here's an example docker-compose.yml file that ..."

File: docs/tools/kompose/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+redis-master                    10.0.202.43   <none>        6379/TCP     4m
+redis-slave                     10.0.1.85     <none>        6379/TCP     4m
+NAME                            READY         STATUS        RESTARTS     AGE
+frontend-2768218532-cs5t5       1/1           Running       0            4m
+redis-master-1432129712-63jn8   1/1           Running       0            4m
+redis-slave-2504961300-nve7b    1/1           Running       0            4m
+```
+Note:
+- You must have a running Kubernetes cluster with a pre-configured kubectl context.
+- Only deployments and services are generated and deployed to Kubernetes. If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead.
```
Comment (janetkuo) [inline docs/tools/kompose/user-guide.md]: These bullet points aren't rendered well on the website.

File: docs/tools/kompose/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+svc/redis-master   172.30.144.56                         <none>        6379/TCP   8s
+svc/redis-slave    172.30.75.245                         <none>        6379/TCP   8s
+NAME               DOCKER REPO                           TAGS          UPDATED
+is/frontend        172.30.12.200:5000/fff/frontend                     
+is/redis-master    172.30.12.200:5000/fff/redis-master                 
+is/redis-slave     172.30.12.200:5000/fff/redis-slave    v1  
+```
+
+Note:
+- You must have a running OpenShift cluster with a pre-configured `oc` context (`oc login`)
```
Comment (janetkuo) [inline docs/tools/kompose/user-guide.md]: here too

File: docs/tools/kompose/user-guide.md (hunk 3 above)
Comment (janetkuo) [inline docs/tools/kompose/user-guide.md]: Change these to `*` to see if it works? Or add a newline after "Note:"

File: docs/tools/kompose/development.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
+---
```
Comment (janetkuo) [inline docs/tools/kompose/development.md]: This doc is for kompose developers. We only put this kind of docs in github repo.

File: docs/tools/kompose/architecture.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
+---
```
Comment (janetkuo) [inline docs/tools/kompose/architecture.md]: This doc is for kompose developers. We only put this kind of docs in github repo.

**Comment by janetkuo** [comment]
@kubernetes/docs @devin-donnelly @pwittrock thoughts on adding kompose docs under "tools"?

**Comment by cdrage** [comment]
Thanks @janetkuo i'll update this and remove the relevant "architecture" and "development" examples, you're right. that should be directed to the github repo instead.

In terms of other pages, I think that the "overview" page, "user guide" page as well as an example / getting started page would be enough of a decent introduction to Kompose.

**Comment by cdrage** [comment]
@janetkuo I've updated the PR with the changes you've suggest.

**Comment by cdrage** [comment]
ping @kubernetes/docs

**Comment by cdrage** [comment]
pingarino @kubernetes/docs @devin-donnelly @pwittrock :)

**Comment by devin-donnelly** [comment]
I'm not a fan of just updating the Tools TOC with Kompose alone--the rest of the tools need to be on equal footing.

@janetkuo, would you mind building out the TOC to include the built-in tools as well as Helm after this is merged?

**Comment by cdrage** [comment]
@janetkuo any changes you wish for me to make? was unfamiliar with the Tools TOC so I added it in manually (is it auto generated?)

**Comment by devin-donnelly** [comment]
@cdrage, I'm okay with adding Kompose to the tools ToC, but we need to add the native Kubernetes tools above it and Helm below it. If we don't do that, it looks like Kompose is the only supported tool, which I don't want.

I can take that on myself, I suppose, since nobody else is jumping to do it.

**Comment by cdrage** [comment]
@devin-donnelly Okay :) Are you able to open an issue and assign yourself to it? Unfortunately I can't do anything assign wise / label wise. 

We can get some Helm PR's up / other third-party tools so Kompose docs isn't lonely up there on the Tools page.

**Comment by devin-donnelly** [comment]
Hi @cdrage , sorry this took so long (got bogged down with 1.5 launch docs), but I've managed to update the Tools TOC and restructure the page slightly in PRs #2075 and #2076 .

This, by necessity, introduced some merge conflicts into tools.yml, but you should be able to resolve those and build on the structure there. Feel free to do so and I'll be happy to merge this PR for you.

**Comment by cdrage** [comment]
@devin-donnelly Not a problem / thanks for everything! :)

I've gone ahead and updated the PR (specifically, removed the conflicting files with `tools.yml`).

**Comment by devin-donnelly** [comment]
@cdrage, feel free to update _tools.yml and the Tools index page to point to your new documentation; I didn't do that because I didn't want to publish the links that wouldn't exist until this PR got merged.

**Comment by cdrage** [comment]
@devin-donnelly Ah crap, so I *was* suppose to keep those changes in. I'll update it.

**Comment by cdrage** [comment]
@devin-donnelly I've updated the index / tools. Let me know if this is correctly done.

**Comment by cdrage** [comment]
@devin-donnelly so I ended up updating it but I'm experiencing a weird issue.

Upon clicking on "Kompose" on the list to the side of the tools index. It takes me to the proper Kompose index file, however, the submenu on the left thinks it's on the "support" tab.

**Comment by janetkuo** [comment]
> Upon clicking on "Kompose" on the list to the side of the tools index. It takes me to the proper Kompose index file, however, the submenu on the left thinks it's on the "support" tab.

You'll need to update `_data/tools.yml` and `docs/tools/index.md` to make kompose docs be linked correctly

**Comment by cdrage** [comment]
@janetkuo Yup, I already did. But I'm not too sure what I'm doing wrong. Can you check the code that's commited in this PR?

File: docs/tools/kompose/index.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
+---
+assignees:
+- cdrage
+
```
Comment (devin-donnelly) [inline docs/tools/kompose/index.md]: Please add an explicit title line here. Something like:

title: Kompose

File: docs/tools/kompose/user-guide.md [hunk 8]
```diff
@@ -0,0 +1,308 @@
+---
+assignees:
+- cdrage
+
```
Comment (devin-donnelly) [inline docs/tools/kompose/user-guide.md]: Please add an explicit title.

title: Kompose User Guide

File: _data/tools.yml [hunk 9]
```diff
@@ -17,5 +17,4 @@ toc:
   - title: Helm
     path: https://github.com/kubernetes/helm
   - title: Kompose
-    path: https://github.com/kubernetes-incubator/kompose
-  
+    path: /docs/tools/kompose/
```
Comment (janetkuo) [inline _data/tools.yml]: I think I know what's causing the problem here. Due to some recent GitHub Pages changes, we no longer specify title here. Please see this example for how to specify titles: [_data/guides.yml](https://github.com/kubernetes/kubernetes.github.io/blob/master/_data/guides.yml#L8) and [docs/whatisk8s.md](https://github.com/kubernetes/kubernetes.github.io/blame/master/docs/whatisk8s.md#L5)

Basically remove `title: Kompose` here and insert title in docs/tools/kompose/index.md

**Pull request merged**